File: src/schemas/api/guilds/Sticker.ts
import { Snowflake } from "../../Identifiers";
import { PartialUser } from "../users";

export type StickersResponse = StickerResponse[];
export interface StickerResponse {
    id: Snowflake;
    pack_id?: Snowflake;
    name: string;
    description: string | null;
    tags: string;
    type: StickerType;
    format_type: StickerFormatType;
    available?: boolean;
    guild_id?: Snowflake;
    // Only filled for Get Guild Sticker(/s)
    user?: PartialUser;
    // Only filled in for sticker packs
    sort_value?: number;
}

export enum StickerType {
    STANDARD = 1,
    GUILD = 2,
}

export enum StickerFormatType {
    PNG = 1,
    APNG = 2,
    LOTTIE = 3,
    GIF = 4,
}

File: extra/admin-api/Models/Spacebar.Models.Generic/Sticker.cs
using System.Text.Json.Serialization;

namespace Spacebar.Models.Generic;

public class Sticker {
    [JsonPropertyName("id"), JsonNumberHandling(JsonNumberHandling.AllowReadingFromString | JsonNumberHandling.WriteAsString)]
    public required long Id { get; set; }

    [JsonPropertyName("name")]
    public string Name { get; set; }

    [JsonPropertyName("description")]
    public string Description { get; set; }

    [JsonPropertyName("tags")]
    public string Tags { get; set; }

    [JsonPropertyName("available")]
    public bool Available { get; set; }

    [JsonPropertyName("type")]
    public StickerType Type { get; set; }

    [JsonPropertyName("format_type")]
    public StickerFormatType FormatType { get; set; }
}

public enum StickerType {
    Standard = 1,
    Guild = 2
}

public enum StickerFormatType {
    Png = 1,
    Apng = 2,
    Lottie = 3,
    Gif = 4
}